packages/forge2d/lib/src/dynamics/joints/joint.dart

```diff
@@ -131,34 +124,23 @@ abstract class Joint {
 
     _color.setFromRGBd(0.5, 0.8, 0.8);
 
-    switch (type) {
-      case JointType.distance:
-        debugDraw.drawSegment(p1, p2, _color);
-        break;
-
-      case JointType.pulley:
-        {
-          final pulley = this as PulleyJoint;
-          final s1 = pulley.getGroundAnchorA();
-          final s2 = pulley.getGroundAnchorB();
-          debugDraw.drawSegment(s1, p1, _color);
-          debugDraw.drawSegment(s2, p2, _color);
-          debugDraw.drawSegment(s1, s2, _color);
-        }
-        break;
-
-      case JointType.friction:
-        debugDraw.drawSegment(x1, x2, _color);
-        break;
-
-      case JointType.constantVolume:
-      case JointType.mouse:
-        // don't draw this
-        break;
-      default:
-        debugDraw.drawSegment(x1, p1, _color);
-        debugDraw.drawSegment(p1, p2, _color);
-        debugDraw.drawSegment(x2, p2, _color);
+    if (this is DistanceJoint) {
+      debugDraw.drawSegment(p1, p2, _color);
+    } else if (this is PulleyJoint) {
+      final pulley = this as PulleyJoint;
+      final s1 = pulley.getGroundAnchorA();
+      final s2 = pulley.getGroundAnchorB();
+      debugDraw.drawSegment(s1, p1, _color);
+      debugDraw.drawSegment(s2, p2, _color);
+      debugDraw.drawSegment(s1, s2, _color);
+    } else if (this is FrictionJoint) {
+      debugDraw.drawSegment(x1, x2, _color);
+    } else if (this is ConstantVolumeJoint || this is MouseJoint) {
+      // Don't draw this joints.
+    } else {
```

Comment:
@spydon what do we think of instead of using this `else` we use an `else if` with the remaining `Joint`s? 

Personally, I think being explicit here will help with clarity and understanding. If you agree I would do this on a different PR. Then, we have the option to either from an `Error` in the else say that the given `Joint` can't be drawn or just ignore it.

However, to be honest, I think this method should be overridable by each `Joint` and defined under its appropriate class. Benefitting from the OOP paradigm. In addition, if someone wants to achieve a custom debug drawing they could by extending `Joint`. Although, right now we don't support this since `world.createJoint` doesn't support extensions of a `Joint` yet.

wdyt?

Reply:
Sounds good, we can do it in two steps then.